File: API/Features/Reservations/Dtos/Read/ReservationBoatReadDto.cs
using API.Infrastructure.Classes;

namespace API.Features.Reservations {

    public class ReservationBoatReadDto {

        public int Id { get; set; }
        public string Description { get; set; }
        public SimpleEntity HullType { get; set; }
        public SimpleEntity Usage { get; set; }
        public SimpleEntity Nationality { get; set; }
        public ReservationBoatInsuranceReadDto Insurance { get; set; }
        public decimal Loa { get; set; }
        public decimal Beam { get; set; }
        public decimal Draft { get; set; }
        public string RegistryPort { get; set; }
        public string RegistryNo { get; set; }

    }

}

File: FrontEnd/src/app/features/reservations/classes/dtos/read/reservation-boat-read-dto.ts
import { ReservationBoatInsuranceReadDto } from './reservation-boat-insurance-read-dto'
import { ReservationBoatNationalityReadDto } from './reservation-boat-nationality-read-dto'
import { SimpleEntity } from 'src/app/shared/classes/simple-entity'

export interface ReservationBoatReadDto {

    reservationId: string
    id: number
    description: string
    hullType: SimpleEntity
    usage: SimpleEntity
    nationality: ReservationBoatNationalityReadDto
    insurance: ReservationBoatInsuranceReadDto
    loa: number
    beam: number
    draft: number
    registryPort: string
    registryNo: string

}
